### DISCUSSION TITLE
Error ingesting to DERIVA: Found 4 errors in datapackage "datapackage.json". First error: There is an extra label "" in header at position "4"

### DISCUSSION THREAD

**Discussion by ekropiwnicki** [nih-cfde/published-documentation#223] [Bug] [answered]
My datapackage validated locally, however, I received the error in the title after submitting the datapackage to DERIVA and I am not sure how to interpret it. My understanding is that there are errors in the "datapackage.json" file, however, I don't think I edited that file and simply downloaded it from here: https://osf.io/e5tc2/

I may be missing obvious errors within the other files, however, the error message makes it difficult to interpret what there is to debug.

Dropbox link to files: https://www.dropbox.com/s/tbjtuzjhjlwg7d1/output.zip?dl=0

**Comment by ACharbonneau** [comment, marked as answer]
There are a couple of different things intersecting here:

- The optional frictonless validator step checks some things, but isn't as robust as our on server validation system, so it is not surprising that a package could validate locally but not pass on the server. 

- You are correct that you should not ever have to edit the JSON file we host on OSF. 

- You found an error in our error handling, the website should have told you which table has the extra header, but didn't. I'll make sure that bug gets to the development team

- The actual error you're seeing is that you have an 4th column in your `assay_type.tsv`, `anatomy.tsv`, `taxonomy.tsv`, and `data_type.tsv` files. We dropped synonyms from all 5 controlled vocabulary tables, so the database doesn't know what to do with those columns. You don't have the synonym column for file_format.tsv (the fifth one) so I'm not sure where to suggest checking your workflow, but my first guess is that you might have an older version of the helper script? Or maybe you built those tables yourself and have an internal script that needs updating. In the short term, just deleting those columns manually should fix the error.

**Reply by ekropiwnicki** [reply]
Great thank you for the informative reply! With the suggested changes, the DERIVA ingestion flow was able to complete successfully.
